text-generation-inference/server/text_generation_server/generator.py

```diff
@@ -326,9 +394,40 @@ def warmup(self, batch: Batch) -> int:
             raise ValueError(
                 f"Inconsistent server configuration: please make sure max-prefill-tokens does not exceed {batch_size} x max-input-length."
             )
-        self.prefill(batch)
-        self.clear()
-        return batch_size * self.model.config.sequence_length
+
+        # Counter-intuitively, now we ignore the input batch. Instead, we create dummy batches to cover all possible
+        # batch sizes and sequence lengths.
+        seq_len = self.model.config.sequence_length
+        bucket_seq_len = take_nearest_length(seq_len)
+        requests = [self._create_dummy_request(seq_len) for _ in range(batch_size)]
+        for _ in reversed(range(batch_size)):
+            # Prefill with different truncate sizes to test all prefill lengths. List is reversed so first longest
+            # sequences are tested and, if there is a memory failure, that will appear sooner.
+            for l in reversed(PREFILL_LENGTHS):
```

Comment:
We will certainly need to cap the `PREFILL_LENGTHS` w.r.t the model's config `max_position_embedding` otherwise it might generate too much and also unsupported lengths

Reply:
good point. For now it will stop at the sequence length passed as parameter, but we can also make this check. I will do it on a future PR.